docs/sections/contributor_guide/developer_setup.rst

```diff
@@ -1,5 +1,183 @@
+your main shell, you can create a ``~/.condevrc`` file, which will be
+sourced by ``make devshell``. When in doubt, you might:
+
+   .. code::
+
+     cat <<EOF >~/.condevrc
+     source ~/.bashrc
+     EOF
+
+
+Using a development shell
+=========================
+
+In an active development shell, the following ``make`` targets are
+available and act on all ``.py`` files under ``src/``:
+
+
++---------------------+------------------------------------------------------------+
+| Command             |  Description                                               |
++=====================+============================================================+
+| ``make format``     | Format with |black|_ and |isort|_                          |
+|                     |                                                            |
+|                     | Formats docstrings and ``.jsonschema`` documents           |
++---------------------+------------------------------------------------------------+
+| ``make lint``       | Lint with ``pylint``                                       |
+|                     |                                                            |
++---------------------+------------------------------------------------------------+
+| ``make type check`` | Typecheck with ``mypy``                                    |
++---------------------+------------------------------------------------------------+
+| ``make unittest``   | Run unit tests and report coverage with ``pytest`` and     |
+|                     | ``coverage``                                               |
++---------------------+------------------------------------------------------------+
+| ``make test``       | Equivalent to                                              |
+|                     | ``make lint && make typecheck && make unittest``           |
+|                     |                                                            |
+|                     | Checks defined CLI scripts                                 |
++---------------------+------------------------------------------------------------+
+
+
+Note that ``make format`` is never run automatically, to avoid
+reformatting under-development code in a way that might surprise the
+developer. A useful development idiom is to periodically run
+``make format && make test`` to perform a full code-quality sweep
+through the code.
+
+The ``make test`` command is also automatically executed when ``conda``
+builds a ``uwtools`` package, so it is important to periodically run
+these tests during development and, crucially, before merging changes,
+to ensure that the tests will pass when CI builds the ``workflow-tools``
+code.
+
+
+The order of the targets above is intentional, and possibly useful:
+
+   * ``make format`` will complain about certain kinds of syntax errors that
+     would cause all the remaining code-quality tools to fail (and may change
+     line numbers reported by other tools, if it ran after them).
```

Comment:
Here, too, each bullet line should be one long line, without artificial line breaks.